### ISSUE TITLE
Remove aboutUrl and valueUrl from metadata specification

### ISSUE THREAD

**Issue by rufuspollock** [w3c/csvw#715]
_This is based on my understanding reading the current spec and may be faulty ;-) - please correct me if i am misunderstanding the purpose of `aboutUrl`_

Why remove?

Primary purpose of `aboutUrl` and `valueUrl` (AFAICT) is to allow the generation of URI based identifiers for rows,  columns and cells.

For most use cases this is of minor importance and could be performed explictly at the processing stage rather than written into the metadata.

At the same time `aboutUrl` adds substantially to the spec especially when combined with URI template properties section (is URI template properties relevant for anything other than `aboutUrl` and `valueUrl`

In summary: value in spec is limited (could achieve this by other means) but it adds substantial implementational and cognitive complexity to the specification.

**Comment by JeniT** [comment]
-1

URI template properties are also relevant for `propertyUrl` which is used to generate property URLs. All three properties are particularly useful and relevant when creating RDF, which is important for many members of the WG.

`aboutUrl` and `valueUrl` also enable the generation of nested/structured JSON and RDF from CSV files, which is a substantial benefit when transformation is your goal.

There are many libraries available supporting URI template parsing so implementation cost is relatively low.

**Comment by rufuspollock** [comment]
@JeniT i can only give my experience which was that i had to re-read these items and the URI-Template portion of the spec several times to grok what was going on. My question is why special-case generation of a particular target output format in a descriptive spec. If we need transformation stuff like this why not put into a dedicated separate mini-spec on transformation rather than have it add complexity to the metadata spec - cf #722

**Comment by danbri** [comment]
-1
(propertyUrl valued too)

These are useful for JSON-LD, RDFa, Microdata, alongside various other more explicitly RDF-oriented formats (Turtle, N-Triples, RDF/XML etc.). But format details aside they are usefully considered metadata because they serve to make more explicit the meaning of the table's content.

**Comment by JeniT** [comment]
@rgrp For sure we need a Primer to help lead people through the spec -- that's one of the things that I hope that we'll create over the next couple of months. The definitions of how the transformations work are in separate specs, but in my opinion it would be even more confusing to have some parts of the metadata syntax defined in one spec and some in another.

**Comment by 6a6d74** [comment]
-1 to Rufus' proposal. Agree with points made by @JeniT and @danbri
previously

**Comment by gkellogg** [comment]
I stated elsewhere the importance of aboutUrl and valueUrl, so -1 to removing.

**Comment by rufuspollock** [comment]
I don't get this - having a primer for a spec doesn't solve the fact that it is overly complex. A primer is useful but this adds RDF-style complexity to a spec that doesn't need it.

I'm wondering if we should trial this with some folks who are non-RDF and not been involved in the spec process to really get a sense of whether they find it easy to understand the spec as is - and portions like this. I personally, found this hard to parse and hard to understand and, to repeat, adds significant complexity.

I guess a fundamental question is the importance of generating RDF from CSVs. For me this is a relatively secondary goal relative to other ones - e.g. enabling widespread use and integration into mainstream developer tools and systems. Obviously if RDF-ization is a primary goal of the spec - vs e.g. ease of use for standard tool creators and users - then these kind of features makes much more sense.

**Comment by JeniT** [comment]
@rgrp Supporting transformation to RDF is one of the chartered objectives of the Working Group. It appeared in several of the use cases that we gathered. It is of critical importance to the majority of the Working Group members, more so than conversion to JSON. So yes, supporting conversion to RDF is a primary goal for the work of this group.

Which means that to address your concerns we have to make it easier to understand rather than remove the functionality. Which are the bits that are hard to grok?

**Comment by iherman** [comment]
I agree with what @JeniT's and @danbri's arguments. 

Additionally: URL templates are _not_ for RDF only, let us not focus on the pros or cons of an RDF conversion may not be fruitful. We are talking about _data on the Web_, and URL-s are integral part of such data. Whether in CSV, JSON, RDF, or any other formats. To be able to generate the right forms of URL-s is therefore vital. Examples like the [one on interrelated tables in the JSON conversion document](http://w3c.github.io/csvw/publishing-snapshots/CR-csv2json/Overview.html#h-example-public-sector-roles-and-salaries) is an example where the URL templates are part of real life examples of usage for a generated JSON.

The group spent quite some time in trying to locate the minimal possible transformation engine without the necessity to come up with something new. The group rejected (roughly a year ago) other proposals, and the mechanism based on an existing RFC for URL templates was judged to be right level of functionality that already has existing usage and implementations and without involving more complex functionalities. I do not see any new argument to reopen this decision.

-1 from me, thus.

**Comment by rufuspollock** [comment]
@iherman my question then is that if transformations such as this are so important could they be in a dedicated transformation spec - so that the large majority of users who do not need them (and won't use them) do not have to parse the added complexity (and likewise tool authors also do not have to implement to have basic compliance with the metadata spec).

I will also flag that generating URLs for things is not that common use case for many (if the majority) of normal data wranglers I come across.

That said, it is worth noting that this feature isn't just about "urls" - it is something that allows you to do rich datatypes albeit in a very RDF / URIs as identifiers type way. In fact, I believe one of the reasons it was first introduced - see https://github.com/w3c/csvw/issues/26. That said, as I made fairly clear in that issue, I thought we were over-complexifying things even then.

**Comment by iherman** [comment]
@rgrp :

> @iherman my question then is that if transformations such as this are so important could they be in a dedicated transformation spec - so that the large majority of users who do not need them (and won't use them) do not have to parse the added complexity (and likewise tool authors also do not have to implement to have basic compliance with the metadata spec).

I think this issue is now separated into #722.

> That said, it is worth noting that this feature isn't just about "urls" - it is something that allows you to do rich datatypes albeit in a very RDF / URIs as identifiers type way. In fact, I believe one of the reasons it was first introduced - see #26. That said, as I made fairly clear in that issue, I thought we were over-complexifying things even then.

The RFC for URL templates, which is used here, has a syntax and semantics that is geared towards URL-s. Ie, this feature is about URL-s and is aimed at URL-s only (of course, it can be misused, but that is true with every feature). 

(#26 was about (RDF) typing; RDF types are URL-s)

**Comment by JeniT** [comment]
Discussed & closed in favour of editorial changes from #722 (i.e. rework transformation into a dedicated section)
